### DISCUSSION TITLE
Play from Youtube error?

### DISCUSSION THREAD

**Discussion by Dacogetul** [openplayerjs/openplayerjs#421] [General]
Hello
I'm trying to set the player to play from YouTube, but it doesn't work. I think that the youtube plugin is separate and should be added. Can you tell me where I can find it?
I tried the standard player version where I only modified the source as it appears in your example on codepen.
My attempt is  [here](https://codepen.io/Dragos-Brojbeanu/pen/rNoLPEm)

**Comment by Dacogetul** [comment]
I find your specific player for that and now i ask if both options work together. I will try now...

**Comment by Dacogetul** [comment]
Not work....for me

**Reply by rafa8626** [reply]
This CodePen works for me: https://codepen.io/rafa8626/pen/wvvOYpg

When you load libraries in CodePen, you need to make sure all of them are using https, not http, because that causes conflicts

**Reply by Dacogetul** [reply]
I tried with my website and it still doesn't work, and in codepen I don't know the location of the library's. I can't see the links in your example either, so I can put them in my codepen.

**Comment by Dacogetul** [comment]
Now i see you load multiple librarys: 
<script type="text/javascript" id="www-widgetapi-script" src="https://www.youtube.com/s/player/1844df69/www-widgetapi.vflset/www-widgetapi.js" async=""></script>
<script src="https://www.youtube.com/iframe_api"></script>
Im not sure if that affect you player but just with your Codepen example and Openplayerjs-youtube not work for me with all sources loaded from https://. like in my Codepen now.

**Reply by rafa8626** [reply]
@Dacogetul you don’t need to load those. There’s a dedicated package for it: https://github.com/openplayerjs/openplayerjs-youtube

Check that link and follow the instructions to set it up. If you need a CDN link to use that plugin, use https://cdn.jsdelivr.net/npm/openplayerjs-youtube@2.3.1/dist/openplayerjs-youtube.min.js

**Reply by Dacogetul** [reply]
I try this without youtube settings on the player.....now show just youtube movie but not player just buttons...... [here](https://codepen.io/Dragos-Brojbeanu/pen/rNoLPEm)

**Reply by rafa8626** [reply]
You didn’t put the CSS styles in the head of the document

**Reply by Dacogetul** [reply]
Yes work great...see there...
Is posible to add youtube plugin and play from playlist both mp4 and youtube using:
<source type="video/mp4" src="video/mp4"></source>
 <source type="video/x-youtube" src="video/x-youtube"></source>
?

**Comment by Dacogetul** [comment]
Yes work great...see there... Is posible to add youtube plugin and play from playlist both mp4 and youtube using:
`<source type="video/mp4" src="video/mp4"></source>
            <source type="video/x-youtube" src="video/x-youtube"></source>`
 ?

Like in [v2-codepen](https://codepen.io/Dragos-Brojbeanu/pen/wvbMdqm)

**Reply by rafa8626** [reply]
@Dacogetul setting the sources inside the video tag are not meant to be used as playlist, but as a way to ensure the browser plays the best match. If you want a playlist, this CodePen has the basis for it: https://codepen.io/rafa8626/pen/GRREQpX

It could work also including YouTube if you already included the player’s YouTube plugin in your site

**Reply by Dacogetul** [reply]
It's not about those settings... I have my own playlist and the javascript code I have additionally (if you remember and which you suspect influences the player) does this only as I see in my v2 using the two separate scripts for mp4 and x-youtube no longer play the movie from youtube, which means that the player cannot automatically choose from among the formats the type of video sent to play.

**Reply by rafa8626** [reply]
If you set the first source as the YouTube source, will it play it? The player is only designed to choose what formats can be played and the order they were set up

**Reply by Dacogetul** [reply]
No..now i put the x-youtube first....if i delete mp4 ....work.

**Reply by Dacogetul** [reply]
I set like [here](https://www.w3schools.com/html/html5_video.asp) but change ogg with x-youtube..

**Reply by rafa8626** [reply]
In your CodePen if you remove the first 2 sources it works because the sources are invalid; if you set a valid MP4 source after the YouTube one it should work, since the MP4 must exist, and in your CodePen the MP4 is considered but nothing happens once the src is not a valid mp4

**Reply by Dacogetul** [reply]
I can't play it on codepen because I need a playlist from which to select the movie...any source I add, I can't tell the player which one to run to see if it works...I have to try it on my server...
